### PR TITLE
CRAFTERCMS-1970

### PR DESCRIPTION
**Pull request by dejan-brkic**
Added sync blueprints feature

### Ticket reference or full description of what's in the PR
https://github.com/craftercms/craftercms/issues/1970

### CODE REVIEW

File: src/main/java/org/craftercms/studio/impl/v1/repository/git/GitContentRepository.java [hunk 1]
```diff
@@ -850,6 +852,31 @@ public void bootstrap() throws Exception {
...
+                } catch (GitAPIException err) {
+                    logger.error("error creating initial commit for global configuration", err);
+                }
+            } else {
+                // rsync blueprints
+                String bootstrapFolderPath = this.ctx.getRealPath(FILE_SEPARATOR + BOOTSTRAP_REPO_PATH + FILE_SEPARATOR + BOOTSTRAP_REPO_GLOBAL_PATH + FILE_SEPARATOR + studioConfiguration.getProperty(BLUE_PRINTS_PATH));
+                Path source = java.nio.file.FileSystems.getDefault().getPath(bootstrapFolderPath);
+                Path globalConfigPath = helper.buildRepoPath(GitRepositories.GLOBAL);
+                Path blueprintsPath = Paths.get(globalConfigPath.toAbsolutePath().toString(), studioConfiguration.getProperty(BLUE_PRINTS_PATH));
+                String[] cmd = new String[] {"rsync", "-avz", "--delete", source.toAbsolutePath().toString() + FILE_SEPARATOR, blueprintsPath.toAbsolutePath().toString() };
```
Comment (sumerjabri) [inline src/main/java/org/craftercms/studio/impl/v1/repository/git/GitContentRepository.java:865]: Will this work on Windows?

Comment (sumerjabri) [reply src/main/java/org/craftercms/studio/impl/v1/repository/git/GitContentRepository.java:865]: How about we delete and recreate the blueprints folder if something has changed?

File: src/main/webapp/repo-bootstrap/global/blueprints/website_editorial/test2.text [hunk 2]
```diff
@@ -0,0 +1,3 @@
+test
```
Comment (sumerjabri) [inline src/main/webapp/repo-bootstrap/global/blueprints/website_editorial/test2.text]: Do we want this file committed?

File: src/main/java/org/craftercms/studio/impl/v1/repository/git/GitContentRepository.java (hunk 1 above)
Comment (sumerjabri) [reply src/main/java/org/craftercms/studio/impl/v1/repository/git/GitContentRepository.java:865]: Perhaps a checksum file or a date file and we check/decide.

**Review by sumerjabri — CHANGES_REQUESTED** [review]

File: src/main/java/org/craftercms/studio/impl/v1/repository/git/GitContentRepository.java (hunk 1 above)
Comment (dejan-brkic) [reply src/main/java/org/craftercms/studio/impl/v1/repository/git/GitContentRepository.java:865]: checksum file or timestamp file is a nice idea, the question is how do we automate it (if we rely on the user to update it, we might get into a problem if he forgets)

Comment (sumerjabri) [reply src/main/java/org/craftercms/studio/impl/v1/repository/git/GitContentRepository.java:865]: the idea is to create a signature-file based on the war file's build number when starting up, we look at that file and if it's different from our build number, we delete the folder for the BPs and create them again and update the file

**Pull request closed without merging**